include/MFRC522_func.h

```diff
@@ -0,0 +1,106 @@
+    // Serial.println(F(" ..."));
+    status = (MFRC522::StatusCode) mfrc522.MIFARE_Read(blockAddr, buffer, &size);
+    if (status != MFRC522::STATUS_OK) {
+        Serial.print(F("MIFARE_Read() failed: "));
+        Serial.println(mfrc522.GetStatusCodeName(status));
+        return false;
+    }
+    // Serial.print(F("Data in block ")); Serial.print(blockAddr); Serial.println(F(":"));
+    // dump_byte_array(buffer, 16); Serial.println();
+    // Serial.println();
+    return true;
+}
+
+bool write_and_verify(byte blockAddr, byte dataBlock[], byte buffer[], byte size) {
+    MFRC522::StatusCode status;
+
+    // Authenticate using key A
+    //Serial.println(F("Authenticating using key A..."));
+    //status = (MFRC522::StatusCode) mfrc522.PCD_Authenticate(MFRC522::PICC_CMD_MF_AUTH_KEY_A, trailerBlock, &key, &(mfrc522.uid));
+    //if (status != MFRC522::STATUS_OK) {
+    //    Serial.print(F("PCD_Authenticate() failed: "));
+    //    Serial.println(mfrc522.GetStatusCodeName(status));
+    //    return false;
+    //}
+    
+    // Write data to the block
+    // Serial.print(F("Writing data into block ")); Serial.print(blockAddr);
+    // Serial.println(F(" ..."));
+    // dump_byte_array(dataBlock, 16); Serial.println();
+    status = (MFRC522::StatusCode) mfrc522.MIFARE_Write(blockAddr, dataBlock, 16);
+    if (status != MFRC522::STATUS_OK) {
+        Serial.print(F("MIFARE_Write() failed: "));
+        Serial.println(mfrc522.GetStatusCodeName(status));
+        return false;
+    }
+    Serial.println();
+
+    // Read data from the block (again, should now be what we have written)
+    // Serial.print(F("Reading data from block ")); Serial.print(blockAddr);
+    // Serial.println(F(" ..."));
+    status = (MFRC522::StatusCode) mfrc522.MIFARE_Read(blockAddr, buffer, &size);
+    if (status != MFRC522::STATUS_OK) {
+        Serial.print(F("MIFARE_Read() failed: "));
+        Serial.println(mfrc522.GetStatusCodeName(status));
+        return false;
+    }
+    // Serial.print(F("Data in block ")); Serial.print(blockAddr); Serial.println(F(":"));
+    // dump_byte_array(buffer, 16); Serial.println();
+
+    // Check that data in block is what we have written
+    // by counting the number of bytes that are equal
+    Serial.println(F("Checking result..."));
+    byte count = 0;
+    for (byte i = 0; i < 16; i++) {
+        // Compare buffer (= what we've read) with dataBlock (= what we've written)
+        if (buffer[i] == dataBlock[i])
+            count++;
+    }
+    Serial.print(F("Number of bytes that match = ")); Serial.println(count);
+    if (count == 16) {
+        Serial.println(F("Success :-)"));
+        return true;
+    } else {
+        Serial.println(F("Failure, no match :-("));
+        Serial.println(F("  perhaps the write didn't work properly..."));
+        Serial.println();
+        return false;
+    }
+}
+
+bool UIDcompare (unsigned int prevUID[], unsigned int currUID[], int UIDLen) {
```

Comment:
is it used anywhere?

Reply:
not right now, but I think we should consider using it to not act on same chip over and over